navigation/index.tsx

```diff
@@ -117,187 +113,189 @@ if (Platform.OS === "android") {
-                    justifyContent: "center",
-                    alignItems: "center",
-                    marginTop: Platform.OS === "ios" ? 35 : 0,
-                  }}
-                >
-                  <TouchableOpacity
-                    activeOpacity={pressedOpacity}
-                    style={{
-                      position: "absolute",
-                      left: 24,
                       alignItems: "center",
-                      justifyContent: "center",
-                      width: 50,
-                      height: 50,
+                      marginTop: Platform.OS === "ios" ? 35 : 0,
                     }}
-                    onPress={() => navigation.goBack()}
                   >
-                    <AntDesign name="left" size={24} color="black" />
-                  </TouchableOpacity>
-                  <Text style={fonts.pageHeading3}>New Listing</Text>
-                  <TouchableOpacity
-                    activeOpacity={pressedOpacity}
+                    <TouchableOpacity
+                      activeOpacity={pressedOpacity}
+                      style={{
+                        position: "absolute",
+                        right: 20,
+                        alignItems: "center",
+                        justifyContent: "center",
+                        height: 50,
+                        width: 50,
+                      }}
+                      onPress={() => navigation.goBack()}
+                    >
+                      <AntDesign
+                        name="close"
+                        size={24}
+                        color="black"
+                        style={{ marginStart: 18 }}
+                      />
+                    </TouchableOpacity>
+                    <Text style={fonts.pageHeading3}>New Listing</Text>
+                  </View>
+                );
+              },
+            })}
+            component={NewPostImage}
+          />
+          <Stack.Screen
+            name="NewPostDetail"
+            options={({ navigation }) => ({
+              headerShadowVisible: false,
+              headerStyle: styles.headerNoShadow,
+              headerBackVisible: false,
+              headerTitleAlign: "center",
+              header: () => {
+                return (
+                  <View
                     style={{
-                      position: "absolute",
-                      right: 20,
-                      alignItems: "center",
+                      height: 70,
+                      backgroundColor: "#ffffff",
                       justifyContent: "center",
-                      width: 50,
-                      height: 50,
+                      alignItems: "center",
+                      marginTop: Platform.OS === "ios" ? 35 : 0,
                     }}
-                    onPress={() => navigation.navigate("Root")}
                   >
-                    <AntDesign name="close" size={24} color="black" />
```

Comment:
This header component appears to have quite a lot of code associated with it -- maybe we can pull this out into a separate component

Reply:
And then you can do things like abstracting out the styles for that component instead of putting the styles directly in the `style` prop

Reply:
Most of the changes to this file are either due to my auto-formatter or previous PRs. The purpose of this PR is to make sure sign in functionality works, I think it makes more sense to have general codebase cleaning in a separate PR, since I'm sure there are tons of other files that don't use best practice that are already in the codebase.